Dashboard/app/js/lib/views/data/inspect-data-table-views.js

```diff
@@ -215,9 +215,33 @@ FLOW.inspectDataTableView = FLOW.View.extend({
 FLOW.DataItemView = FLOW.View.extend({
   tagName: 'span',
   deleteSI: function () {
-    var SI;
+    var SI,slKey, sl;
     SI = FLOW.store.find(FLOW.SurveyInstance, this.content.get('keyId'));
     if (SI !== null) {
+    	// check if we also have the data point loaded locally
+    	slKey = SI.get('surveyedLocaleId');
+    	SL = FLOW.store.filter(FLOW.SurveyedLocale,function(item){
+    		return item.get('keyId') == slKey;
+    	});
+    	// if we have found the surveyedLocale, check if there are more 
+    	// formInstances inside it
+    	if (!Ember.empty(SL)){
+    		// are there any other formInstances loaded for this surveyedLocale?
+    		// if not, we also need to not show the locale any more. 
+    		// it will also be deleted automatically in the backend,
+    		// so this is just to not show it in the UI
+    		SiList = FLOW.store.filter(FLOW.SurveyInstance,function(item){
```

Comment:
Will this guarantee that all `SurveyInstance` objects that match the criteria are considered?